core/src/avm2/parameters.rs

```diff
@@ -107,25 +106,21 @@ impl<'gc> ParametersExt<'gc> for &[Value<'gc>] {
         name: &'static str,
     ) -> Result<Object<'gc>, Error<'gc>> {
         match self[index] {
-            Value::Null | Value::Undefined => Err(make_error_2007(activation, name)),
+            Value::Null => Err(make_error_2007(activation, name)),
             Value::Object(o) => Ok(o),
-            primitive => Ok(PrimitiveObject::from_primitive(primitive, activation)
-                .expect("Primitive object is infallible at this point")),
+            _ => panic!("get_object should read Object or Null"),
         }
     }
 
     fn try_get_object(
         &self,
-        activation: &mut Activation<'_, 'gc>,
+        _activation: &mut Activation<'_, 'gc>,
         index: usize,
     ) -> Option<Object<'gc>> {
         match self[index] {
-            Value::Null | Value::Undefined => None,
+            Value::Null => None,
             Value::Object(o) => Some(o),
-            primitive => Some(
-                PrimitiveObject::from_primitive(primitive, activation)
-                    .expect("Primitive object is infallible at this point"),
-            ),
+            _ => panic!("try_get_object should read Object or Null"),
```

Comment:
why not keep the `Undefined` variant support?
As is, this mostly just forced changing a bunch of `try_get_object` to more verbose `get_value().as_object()`. (and also feels out of scope?)